### PR TITLE
Integration sql

### PR DESCRIPTION
**Pull request by Leen-odeh3**

### CODE REVIEW

File: Simple Inventory Management System/Program.cs [hunk 1]
```diff
@@ -1,86 +1,102 @@
...
+class Program
 {
-    class Program
+    static async Task Main(string[] args)
     {
-        static void Main(string[] args)
-        {
-            Inventory inventory = new Inventory();
+        var config = new ConfigurationBuilder()
+            .AddJsonFile(@"appsettings.json", optional: false, reloadOnChange: true)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: Why interpolated string?

Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: Why interpolated string?

File: Simple Inventory Management System/Program.cs [hunk 2]
```diff
@@ -1,86 +1,102 @@
...
-                Console.WriteLine("4. Delete Product");
-                Console.WriteLine("5. Search for Product");
-                Console.WriteLine("6. Exit");
+        var connectionString = config.GetConnectionString("Default");
 
-                Console.ForegroundColor = ConsoleColor.Yellow;
-                Console.Write("Enter your choice: ");
-                Console.ForegroundColor = ConsoleColor.White;
-                string choice = Console.ReadLine();
+        var productRepository = new ProductRepository(connectionString);
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs:16]: You used to do dependency injection;I will not ask you to do it here, but this is a reminder to use it in the future

File: Simple Inventory Management System/Program.cs [hunk 3]
```diff
@@ -1,86 +1,102 @@
...
 
+        try
+        {
+            Console.WriteLine("Welcome to the Simple Inventory Management System!");
+            Console.WriteLine("Database connected successfully");
+            string answer;
+            do
+            {
+                Console.WriteLine("\nSelect from the options\n1.Add Product\n2.Display Products\n3.Edit Product\n4.Delete Product\n5.Search Product");
+                int choice = int.Parse(Console.ReadLine());
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: please use `var` when no need to declare the type, in this project and the future ones

File: Simple Inventory Management System/Program.cs [hunk 4]
```diff
@@ -1,86 +1,102 @@
...
-
-                        decimal price;
-                        while (!decimal.TryParse(Console.ReadLine(), out price))
+                    case 1:
+                        // Add Product
+                        Console.WriteLine("Enter item name:");
+                        var itemName = Console.ReadLine();
+                        Console.WriteLine("Enter item price:");
+                        decimal itemPrice;
+                        while (!decimal.TryParse(Console.ReadLine(), out itemPrice) || itemPrice < 0)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]:
```suggestion
                        while (!decimal.TryParse(Console.ReadLine(), out decimal itemPrice) || itemPrice < 0)
```

File: Simple Inventory Management System/Program.cs [hunk 5]
```diff
@@ -1,86 +1,102 @@
...
-                            Console.WriteLine("Invalid input. Please enter a valid decimal value for the price.");
-                            Console.Write("Enter product price: ");
+                            Console.WriteLine("Invalid price. Please enter a positive number.");
                         }
-                        int quantity;
-                        Console.Write("Enter product quantity: ");
-                        while (!int.TryParse(Console.ReadLine(), out quantity))
+                        Console.WriteLine("Enter item quantity:");
+                        int itemQuantity;
+                        while (!int.TryParse(Console.ReadLine(), out itemQuantity) || itemQuantity < 0)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: same as above

File: Simple Inventory Management System/Program.cs [hunk 6]
```diff
@@ -1,86 +1,102 @@
...
 
                     default:
-                        Console.WriteLine("Invalid choice. Please try again.");
+                        Console.WriteLine("Please enter a valid choice");
                         break;
                 }
-            }
+                Console.WriteLine("\nDo you want to continue? (Yes/No)");
+                answer = Console.ReadLine();
+            } while (answer.ToUpper() == "YES" || answer.ToUpper() == "Y");
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: I don't remember when, but I remember that I have told you about using comparison like:
`string.Equals(val, "YES", StringComparison.OrdinalIgnoreCase)`

File: Simple Inventory Management System/Repositories/ProductRepository.cs [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        }
+        catch (SqlException ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed operation: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+    private async Task DisplayProductListAsync(string query, Dictionary<string, object>? parameters = null)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Repositories/ProductRepository.cs]: The repo has one task, interact with the database not printing or doing anything else

File: Simple Inventory Management System/Repositories/ProductRepository.cs [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"Failed operation: {ex.Message}");
+            Console.ForegroundColor = ConsoleColor.White;
+        }
+    }
+
+
+    private async Task<SqlDataReader> ExecuteReaderAsync(string query, Dictionary<string, object>? parameters = null)
+    {
+        var connection = new SqlConnection(connectionString);
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Repositories/ProductRepository.cs:149]: assume many reads at the same time, does your code support this case? Search for more better approach

File: Simple Inventory Management System/Repositories/ProductRepository.cs [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+        var command = new SqlCommand(query, connection);
+        if (parameters != null)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Repositories/ProductRepository.cs]: use `is` instead of `!=` 
mor readable

**Review by ahmad-m-abbas — CHANGES_REQUESTED** [review]

File: Simple Inventory Management System/Program.cs [hunk 10]
```diff
@@ -8,7 +8,7 @@ class Program
     static async Task Main(string[] args)
     {
         var config = new ConfigurationBuilder()
-            .AddJsonFile(@"appsettings.json", optional: false, reloadOnChange: true)
+            .AddJsonFile("C:\\Users\\hp\\Desktop\\New folder\\Simple Inventory Management System\\appsettings.json", optional: false, reloadOnChange: true)
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Program.cs]: why? my question is why you have used "@"

**Review by ahmad-m-abbas — CHANGES_REQUESTED** [review]

File: Simple Inventory Management System/Program.cs (hunk 10 above)
Comment (Leen-odeh3) [reply Simple Inventory Management System/Program.cs]: both sides true! , I am not add "\\"

Comment (ahmad-m-abbas) [reply Simple Inventory Management System/Program.cs]: My meaning is :
```csharp
.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```

File: Simple Inventory Management System/Repositories/ProductRepository.cs [hunk 11]
```diff
@@ -144,49 +144,21 @@ private async Task ExecuteNonQueryAsync(string query, Dictionary<string, object>
...
-    }
-
-
     private async Task<SqlDataReader> ExecuteReaderAsync(string query, Dictionary<string, object>? parameters = null)
     {
         var connection = new SqlConnection(connectionString);
         var command = new SqlCommand(query, connection);
 
-        if (parameters != null)
+        if (!(parameters is null))
```
Comment (ahmad-m-abbas) [inline Simple Inventory Management System/Repositories/ProductRepository.cs:152]: is not

**Pull request merged**